xls/modules/zstd/memory/BUILD

```diff
@@ -0,0 +1,714 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@rules_hdl//place_and_route:build_defs.bzl", "place_and_route")
+load("@rules_hdl//synthesis:build_defs.bzl", "benchmark_synth", "synthesize_rtl")
+load("@rules_hdl//verilog:providers.bzl", "verilog_library")
+load("@xls_pip_deps//:requirements.bzl", "requirement")
+load(
+    "//xls/build_rules:xls_build_defs.bzl",
+    "xls_benchmark_ir",
+    "xls_dslx_library",
+    "xls_dslx_test",
+    "xls_dslx_verilog",
+)
+
+package(
+    default_applicable_licenses = ["//:license"],
+    default_visibility = ["//xls:xls_users"],
+    licenses = ["notice"],
+)
+
+xls_dslx_library(
+    name = "axi_dslx",
+    srcs = ["axi.x"],
+)
+
+xls_dslx_library(
+    name = "axi_st_dslx",
+    srcs = ["axi_st.x"],
+)
+
+xls_dslx_library(
+    name = "common_dslx",
+    srcs = [
+        "common.x",
+    ],
+    deps = [":axi_dslx"],
+)
+
+xls_dslx_test(
+    name = "common_dslx_test",
+    library = ":common_dslx",
+    tags = ["manual"],
+)
+
+CLOCK_PERIOD_PS = "750"
+# Clock periods for modules that exceed the 750ps critical path in IR benchmark
+AXI_READER_CLOCK_PERIOD_PS = "1800"
+AXI_STREAM_REMOVE_EMPTY_CLOCK_PERIOD_PS = "1300"
+MEM_READER_CLOCK_PERIOD_PS = "2600"
+
+common_codegen_args = {
+    "delay_model": "asap7",
+    "reset": "rst",
+    "worst_case_throughput": "1",
+    "use_system_verilog": "false",
+    "clock_period_ps": CLOCK_PERIOD_PS,
+    "clock_margin_percent": "20",
+    "multi_proc": "true",
+}
+
+xls_dslx_library(
+    name = "axi_reader_dslx",
+    srcs = ["axi_reader.x"],
+    deps = [
+        ":axi_dslx",
+        ":axi_st_dslx",
+        ":common_dslx",
+    ],
+)
+
+xls_dslx_test(
+    name = "axi_reader_dslx_test",
+    library = ":axi_reader_dslx",
+    tags = ["manual"],
+)
+
+axi_reader_codegen_args = common_codegen_args | {
+    "module_name": "axi_reader",
+    "pipeline_stages": "2",
+    "streaming_channel_data_suffix": "_data",
+    "flop_inputs_kind": "skid",
+    "flop_outputs_kind": "skid",
+    "clock_period_ps": AXI_READER_CLOCK_PERIOD_PS,
+}
+
+xls_dslx_verilog(
+    name = "axi_reader_verilog",
+    codegen_args = axi_reader_codegen_args,
+    dslx_top = "AxiReaderInst",
+    library = ":axi_reader_dslx",
+    tags = ["manual"],
+    verilog_file = "axi_reader.v",
+)
+
+xls_benchmark_ir(
+    name = "axi_reader_opt_ir_benchmark",
+    src = ":axi_reader_verilog.opt.ir",
+    benchmark_ir_args = axi_reader_codegen_args | {
+        "pipeline_stages": "10",
+        "top": "__axi_reader__AxiReaderInst__AxiReader_0__16_32_4_4_4_3_2_14_next",
```

Comment:
maybe use https://google.github.io/xls/bazel_rules_macros/#get_mangled_ir_symbol?

Reply:
ah you're right:
>       is_proc_next: A boolean flag denoting whether the symbol is a
>         next proc function. The argument is mutually exclusive with arguments:
>        'parametric_values' and 'is_implicit_token'.

We should file a feature request for this to be supported, one workaround I can see is to define wrapper proc rather than putting the parameters in the BUILD file.